Lib/email/_header_value_parser.py

```diff
@@ -2713,16 +2713,23 @@ def _fold_as_ew(to_encode, lines, maxlen, last_ew, ew_combine_allowed, charset):
         trailing_wsp = to_encode[-1]
         to_encode = to_encode[:-1]
     new_last_ew = len(lines[-1]) if last_ew is None else last_ew
+
+    encode_as = 'utf-8' if charset == 'us-ascii' else charset
+
+    # The RFC2047 chrome takes up 7 characters plus the length
+    # of the charset name.
+    chrome_len = len(encode_as) + 7
+
+    if (chrome_len + 1) >= maxlen:
+        raise ValueError("Maximum length too short to fit any values")
```

Comment:
I picked `ValueError` here, but if there's a better error type let me know, I'm not super familiar with the `email` module.

Reply:
We should probably use `email.errors.HeaderParseError` instead of `ValueError`. 

It would be nice if you could use single quotes for consistency in the module. 

Also, the error message seems a bit vague given that it is not too short to fit *any* value, but too short to hold encoded word. An ascii value would be just fine with the line length).  And, `Maximum length` is a hard to make sense when coming from an exception message. `max_line_length` is more greppable to me.
What do you think about this error?

 `max_line_length is too small to fit an encoded word`.

Reply:
+1 for `HeaderParseError` but @bitdancer has more up-to-date perspective than I might have.

Reply:
No, a ValueError would be more appropriate here, I think, because it is an error in the *value* of maxlen.  This isn't a *parsing* error, it's a *rendering* error.

Now, that said you might want to consider the fact that in `_fold_mime_parameters` I deal with this issue by bumping maxlen to 78 rather than raising an error.  I'm not sure that was the right choice,  but whatever we do, it should probably be made consistent between the two cases.  It is so sad that I never came back to fix that XXX comment I left myself :(

Note: I unfortunately am not too likely to see github comments, I get too much noise from that source...I just caught this one by chance.  If you need my attention please post on the issue.

Reply:
> Also, the error message seems a bit vague given that it is not too short to fit _any_ value, but too short to hold encoded word. An ascii value would be just fine with the line length). And, `Maximum length` is a hard to make sense when coming from an exception message. `max_line_length` is more greppable to me.

I agree that "Maximum length" is not going to be a good error, switching to `max_line_length` is a good idea.

Regarding the "encoded word" situation, I guess when I wrote this I had assumed that the RFC chrome was added to every line regardless, but only realized that that wasn't the case through testing. I'm still not entirely clear *why* the encoding chrome is unnecessary for shorter words, but I think the suggested comments at least explain *when* the chrome becomes necessary.